internal/controller/cloudflareaccessapplication_controller_test.go

```diff
@@ -438,5 +438,56 @@ var _ = Describe("CloudflareAccessApplication controller", Ordered, func() {
 			Expect(err).To(Not(HaveOccurred()))
 			Expect(cfResource.Name).To(Equal(found.Spec.Name))
 		})
+
+		It("should successfully reconcile CloudflareAccessApplication whose AccessApplicationID references a missing Application", func() {
+			By("Recreating the custom resource for the Kind CloudflareAccessApplication")
+			typeNamespaceName := types.NamespacedName{Name: "cloudflare-app-seven", Namespace: cloudflareName}
+
+			previousCreatedAndUpdatedDate := metav1.NewTime(time.Now().Add(-time.Hour * 24))
+			apps := &v1alpha1.CloudflareAccessApplication{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      typeNamespaceName.Name,
+					Namespace: namespace.Name,
+				},
+				Spec: v1alpha1.CloudflareAccessApplicationSpec{
+					Name:   "missing application",
+					Domain: "recreate-application.cf-operator-tests.uk",
+				},
+				Status: v1alpha1.CloudflareAccessApplicationStatus{
```

Comment:
I think we have to use k8sClient.Status.Update() to update the status. https://github.com/kubernetes-sigs/controller-runtime/issues/2850